pom-gwt.xml

```diff
@@ -0,0 +1,277 @@
+      <resource>
+        <directory>src/</directory>
+        <excludes>
+          <exclude>**/*.java</exclude>
+        </excludes>
+        <includes>
+          <include>**/*</include>
+        </includes>
+        <filtering>true</filtering>
+      </resource>
+    </resources>
+
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>add-source</id>
+            <phase>generate-sources</phase>
+            <goals><goal>add-source</goal></goals>
+            <configuration>
+              <sources>
+                <source>gen</source>
+              </sources>
+            </configuration>
+          </execution>
+          <execution>
+            <id>add-test-source</id>
+            <phase>generate-test-sources</phase>
+            <goals><goal>add-test-source</goal></goals>
+            <configuration>
+              <sources>
+                <source>src/com/google/javascript/rhino/testing</source>
+                <source>src/com/google/javascript/jscomp/testing</source>
+                <source>src/com/google/javascript/refactoring/testing</source>
+              </sources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-resources-plugin</artifactId>
+
+        <!-- Move Messages.properties where ScriptRuntime.java expects it. -->
+        <executions>
+          <execution>
+            <id>rhino_ast-custom</id>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${basedir}/target/rhino_ast</outputDirectory>
+              <resources>
+                <resource>
+                  <targetPath>${basedir}/target/classes</targetPath>
+                  <directory>src</directory>
+                  <includes>
+                    <include>com/google/javascript/rhino/Messages.properties</include>
+                  </includes>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <excludes>
+            <exclude>**/gwt/**</exclude>
+            <exclude>**/testing/**</exclude>
+            <exclude>**/webservice/**</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>net.ltgt.gwt.maven</groupId>
+        <artifactId>gwt-maven-plugin</artifactId>
+        <version>1.0-rc-4</version>
+        <extensions>true</extensions>
+        <configuration>
+	  <moduleName>com.google.JsComp</moduleName>
```

Comment:
You should add

```xml
<moduleShortName>jscomp</moduleShortName>
```

(if you want incremental builds to work – i.e. gwt-maven-plugin detecting whether the compiled GWT app is uptodate or stale and decide whether to recompile it – without `moduleShortName`, the plugin will look for a `com.google.JsComp/com.google.JsComp.nocache.js` instead of `jscomp/jscomp.nocache.js` and will end up unconditionally recompiling the gwt-app, as if you'd set [`gwt.forceCompilation`](https://tbroyer.github.io/gwt-maven-plugin/compile-mojo.html#forceCompilation) to `true`)